doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,234 @@
+
+will terminate (with `VERIFICATION SUCCESSFUL`).
+
+
+\subsection jbmc-manual-user-assert User assertions
+
+Elaborating on the example above, users may provide their own assertions, that
+JBMC will try do refute. On line 7, we check the assertion that all odd 
+numbers greater than 1 are prime. To be sure that this always holds, we run 
+JBMC on the example, with a reasonable `unwind` value:
+
+    jbmc Example3.class --function tutorial.Example3.doSomething --unwind 10
+
+Unsurprisingly JBMC doesn't agree, and prints an assertion failure
+(truncated here for readability):
+
+    [...doSomething:(I)V.assertion.1] line 7 assertion at file tutorial/Example3.java: FAILURE
+
+
+Rerunning the analysis with the `--trace` option, we can see find out which
+input value JBMC found to trigger the violation:
+
+    INPUT inputVal: 15
+
+The value chosen by JBMC is arbitrary, and could as well be 9
+or 2084569161.
+Note that to see the original parameter names in the trace 
+output, the java source must be compiled in debug mode (`javac -g`).
+
+\subsection jbmc-manual-library Java Library support
+
+The examples above only involve primitive types. In order to analyse code
+involving Java Library classes, the core models (located in the same
+repository) need to be added to the classpath:
+
+    --cp <path_to_cbmc>/jbmc/src/java_bytecode/library/core-models.jar
+    
+
+The core models library simulate the behaviour of the Java Library and
+contains some commonly used classes, such as Object, Class, String, Enum.
+Some JBMC-specific utilities reside in the `org.cprover` package.
+
+Using the models, JBMC can reason with a variety of the String operations
+from `java.lang.String`, e.g.
+```
+package tutorial;
+
+public class Example4 {
+
+    public static void stringOp() {
+        StringBuilder sb = new StringBuilder("abcd");
+        StringBuilder sb2 = sb.insert(2, "$");
+        assert sb2.toString().startsWith("abc");
+    }
+
+}
+```
+The following command line (note that the current directory is also added to
+the classpath):
+
+    jbmc Example4.class --function tutorial.Example4.stringOp --cp <path_to_cbmc>/jbmc/src/java_bytecode/library/core-models.jar:.
+
+will flag this violation (truncated):
+
+    [java::tutorial.Example4.stringOp:()V.assertion.1] line 8 assertion: FAILURE
+
+Again, the trace shows the string violating the condition in the assertion:
+
+    dynamic_object2={ 'a', 'b', '$', 'c', 'd' }
+
+
+\subsection jbmc-manual-exceptions Exceptions
+
+In its analysis, JBMC can take into account exceptions thrown by user code or
+library classes, as well as runtime exceptions such as NullPointerException.
+To have JBMC reliably notify about all uncaught exceptions, the
+`--throw-runtime-exceptions` can be used.
+
+In the following code, only with `--throw-runtime-exceptions` will JBMC
+signal that the assertion on line 11 can be violated. Without that option,
+line 11 will be considered as unreachable, and hence the assertion to be valid.
```

Comment:
It maybe nice to tell how jbmc thinks the line is unreachable from the log. Also without `--throw-runtime-exception`, I still get different Failure
```[java::tutorial.Example5.exceptionHandling:(Ljava/lang/String;)I.null-pointer-exception.1] line 6 Null pointer check: FAILURE
```

Reply:
You can't actually see that in the output. JBMC doens't tell us which parts of the bytecode it managed to reach.